app/controllers/idv/hybrid_mobile/document_capture_controller.rb

```diff
@@ -3,8 +3,141 @@ module HybridMobile
     class DocumentCaptureController < ApplicationController
       include HybridMobileConcern
 
+      before_action :check_valid_document_capture_session
+      before_action :override_csp_to_allow_acuant
+
       def show
-        # TODO
+        analytics.idv_doc_auth_document_capture_visited(**analytics_arguments)
+
+        Funnel::DocAuth::RegisterStep.new(document_capture_user.id, sp_session[:issuer]).
+          call('document_capture', :view, true)
+
+        render :show, locals: extra_view_variables
+      end
+
+      def update
+        handle_stored_result
+
+        analytics.idv_doc_auth_document_capture_submitted(**analytics_arguments)
+
+        Funnel::DocAuth::RegisterStep.new(document_capture_user.id, sp_session[:issuer]).
+          call('document_capture', :update, true)
+
+        redirect_to idv_hybrid_mobile_capture_complete_url
+      end
+
+      def extra_view_variables
+        # Used to call :verify_document_status in idv/shared/_document_capture.html.erb
+        # That code can be updated after the hybrid flow is out of the FSM, and then
+        # this can be removed.
+        @step_url = :idv_capture_doc_step_url
+
+        url_builder = ImageUploadPresignedUrlGenerator.new
+
+        {
+          flow_path: 'hybrid',
+          document_capture_session_uuid: document_capture_session_uuid,
+          front_image_upload_url: url_builder.presigned_image_upload_url(
+            image_type: 'front',
+            transaction_id: document_capture_session_uuid,
+          ),
+          back_image_upload_url: url_builder.presigned_image_upload_url(
+            image_type: 'back',
+            transaction_id: document_capture_session_uuid,
+          ),
+        }.merge(
+          native_camera_ab_testing_variables,
+          acuant_sdk_upgrade_a_b_testing_variables,
+          in_person_cta_variant_testing_variables,
+        )
+      end
+
+      private
+
+      def analytics_arguments
+        {
+          flow_path: 'hybrid',
+          step: 'document_capture',
+          analytics_id: 'Doc Auth',
+          irs_reproofing: irs_reproofing?,
+        }.merge(**acuant_sdk_ab_test_analytics_args)
+      end
+
+      # @param [DocAuth::Response,
+      #   DocumentCaptureSessionAsyncResult,
+      #   DocumentCaptureSessionResult] response
+      def extract_pii_from_doc(response)
+        pii_from_doc = response.pii_from_doc.merge(
+          uuid: effective_user.uuid,
+          phone: effective_user.phone_configurations.take&.phone,
+          uuid_prefix: ServiceProvider.find_by(issuer: sp_session[:issuer])&.app_id,
+        )
+
+        track_document_state(pii_from_doc[:state])
+      end
+
+      # copied from Flow::Failure module
+      def failure(message, extra = nil)
```

Comment:
Just above this I added [successful_response](https://github.com/18F/identity-idp/blob/0a1bd4209f6c0d9ee34e698ad3fc63ee01a87ec7/app/controllers/idv/document_capture_controller.rb#L192-L194) to the desktop DocumentCaptureController and eventually [merged it into analytics](https://github.com/18F/identity-idp/blob/0a1bd4209f6c0d9ee34e698ad3fc63ee01a87ec7/app/controllers/idv/document_capture_controller.rb#L25) in the update method. You'll probably need to do the same to make the analytics match.

Reply:
Addressed in [9624023](https://github.com/18F/identity-idp/pull/8243/commits/9624023bb36729e5842d3c702b97ddcfd6972cf2)